Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -281,6 +365,125 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Multiple Collections Push & Pull S
+        }
+        for (int d = 1; d <= docCount; ++d) {
+            constexpr size_t bufSize = 80;
+            char docID[bufSize];
+            snprintf(docID, bufSize, "%.*s%d", SPLAT(remotePrefix), d);
+            createFleeceRev(_collections[i], slice(docID), nullslice, bodyWithChannel);
+        }
+    }
+
+    {
+        // Send the docs to remote
+        ReplParams replParams { _collectionSpecs };
+        replParams.setPushPull(kC4OneShot, kC4Disabled);
+        replicate(replParams);
+    }
+
+    deleteAndRecreateDB();
+    initTest({Lavenders, Roses, Tulips}, {channelID});
+
+    // add local docs to Push and Push/Pull collections
+    for (size_t i = 0; i < _collectionCount; ++i) {
+        if (i == iPull) {
+            continue;
+        }
+        for (int d = 1; d <= docCount; ++d) {
+            constexpr size_t bufSize = 80;
+            char docID[bufSize];
+            snprintf(docID, bufSize, "%.*s%d", SPLAT(localPrefix), d);
+            createFleeceRev(_collections[i], slice(docID), nullslice, bodyWithChannel);
+        }
+    }
+
+    {
+        ReplParams replParams { _collectionSpecs };
+        replParams.collections[iPush].push = kC4OneShot;
+        replParams.collections[iPull].pull = kC4OneShot;
+        replParams.collections[iPushPull].push = kC4OneShot;
+        replParams.collections[iPushPull].pull = kC4OneShot;
+        replicate(replParams);
+    }
+
+    auto check = [&]() {
+        for (size_t i = 0; i < _collectionCount; ++i) {
+            c4::ref<C4DocEnumerator> e = c4coll_enumerateAllDocs(_collections[i],
+                                                                 nullptr, ERROR_INFO());
+            unsigned total = 0;
+            unsigned local = 0;
+            unsigned remote = 0;
+            while (c4enum_next(e, ERROR_INFO())) {
+                C4DocumentInfo info;
+                c4enum_getDocumentInfo(e, &info);
+                slice docID_sl {info.docID};
+                total++;
+                if (docID_sl.hasPrefix(localPrefix)) {
+                    local++;
+                }
+                if (docID_sl.hasPrefix(remotePrefix)) {
+                    remote++;
+                }
+            }
+            switch (i) {
+                case iPush:
+                    CHECK(total == docCount);
+                    CHECK(local == docCount);
+                    break;
+                case iPull:
+                    CHECK(total == docCount);
+                    CHECK(remote == docCount);
+                    break;
+                case iPushPull:
+                    CHECK(total == 2 * docCount);
+                    CHECK(local == docCount);
+                    CHECK(remote == docCount);
+                    break;
+                default:
+                    break;
+            }
+        }
+    };
+    check();
+
+    // Clear the local database and check again.
+    deleteAndRecreateDB();
+    initTest({Lavenders, Roses, Tulips}, {channelID});
```

Comment:
Same here as previous comment. Just a suggestion though, isn't necessary to change.